components/form/input/src/Mask/index.js

```diff
@@ -0,0 +1,47 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import IMask from 'imask'
+import Input from '../Input'
+
+class MaskInput extends Component {
+  componentDidMount() {
+    const {mask} = this.props
+    this.mask = new IMask(this.field, mask)
+  }
+
+  componentWillUnmount() {
+    this.mask.destroy()
+  }
+
+  onChange = ({ev, value}) => {
+    const {onChange} = this.props
+    onChange && onChange({value: ev.target.value, target: ev.target})
+  }
+
+  render() {
+    const {name} = this.props
+    return (
+      <Input
+        id={name}
+        reference={input => {
+          this.field = input
+        }}
+        onChange={this.onChange}
```

Comment:
if someone passes onChange prop you onChange handler will not be executed, right?

Reply:
which onChange handler? I'm not sure what you mean